### ISSUE TITLE
Wrong Shape Signature Output after Conversion to TFLite

### ISSUE THREAD

**Issue by AntoineChauviere** [tensorflow/tensorflow#57946]
Hello,

Having problems with the conversion of a keras model to tflite (CNN + BLSTM) I replaced the recursive layers by simple Dense. The model works and converges well.

Here is the summary:

==================================================================================================
 image (InputLayer)             [(None, 32, 128, 3)  0           []                               
                                ]                                                                 
                                                                                                  
 conv2d_7 (Conv2D)              (None, 32, 128, 32)  896         ['image[0][0]']                  
                                                                                                  
 max_pooling2d_4 (MaxPooling2D)  (None, 16, 64, 32)  0           ['conv2d_7[0][0]']               
                                                                                                  
 conv2d_8 (Conv2D)              (None, 16, 64, 64)   18496       ['max_pooling2d_4[0][0]']        
                                                                                                  
 max_pooling2d_5 (MaxPooling2D)  (None, 8, 64, 64)   0           ['conv2d_8[0][0]']               
                                                                                                  
 conv2d_9 (Conv2D)              (None, 8, 64, 128)   73856       ['max_pooling2d_5[0][0]']        
                                                                                                  
 conv2d_10 (Conv2D)             (None, 8, 64, 128)   147584      ['conv2d_9[0][0]']               
                                                                                                  
 max_pooling2d_6 (MaxPooling2D)  (None, 4, 64, 128)  0           ['conv2d_10[0][0]']              
                                                                                                  
 conv2d_11 (Conv2D)             (None, 4, 64, 256)   295168      ['max_pooling2d_6[0][0]']        
                                                                                                  
 batch_normalization_2 (BatchNo  (None, 4, 64, 256)  1024        ['conv2d_11[0][0]']              
 rmalization)                                                                                     
                                                                                                  
 conv2d_12 (Conv2D)             (None, 4, 64, 256)   590080      ['batch_normalization_2[0][0]']  
                                                                                                  
 batch_normalization_3 (BatchNo  (None, 4, 64, 256)  1024        ['conv2d_12[0][0]']              
 rmalization)                                                                                     
                                                                                                  
 max_pooling2d_7 (MaxPooling2D)  (None, 2, 64, 256)  0           ['batch_normalization_3[0][0]']  
                                                                                                  
 conv2d_13 (Conv2D)             (None, 1, 63, 64)    65600       ['max_pooling2d_7[0][0]']        
                                                                                                  
 lambda_1 (Lambda)              (None, 63, 64)       0           ['conv2d_13[0][0]']              
                                                                                                  
 dense_crazy_1 (Dense)          (None, 63, 256)      16640       ['lambda_1[0][0]']               
                                                                                                  
 dropout_1 (Dropout)            (None, 63, 256)      0           ['dense_crazy_1[0][0]']          
                                                                                                  
 dense_crazy_2 (Dense)          (None, 63, 128)      32896       ['dropout_1[0][0]']              
                                                                                                  
 label (InputLayer)             [(None, None)]       0           []                               
                                                                                                  
 dense (Dense)                  (None, 63, 48)       6192        ['dense_crazy_2[0][0]']          
                                                                                                  
 ctc_loss (CTCLayer)            (None, 63, 48)       0           ['label[0][0]',                  
                                                                  'dense[0][0]']                  
                                                                                                  
==================================================================================================

On the other hand, if I keep a part (before CTC) and I convert to tflite, I have an output dimension error, I would like to have [1, 63, 48] but I have this : 

'shape': array([ 1, 1, 48], dtype=int32),
'shape_signature': array([-1, -1, 48], dtype=int32),

CODE : 

prediction_model = keras.models.Model(
    model.get_layer(name="image").input, model.get_layer(name="dense").output
)

tf.saved_model.save(prediction_model, "prediction_model")
loaded = tf.saved_model.load("prediction_model")
infer = loaded.signatures["serving_default"]
converter = tf.lite.TFLiteConverter.from_saved_model("prediction_model") # path to the SavedModel directory
tflite_model = converter.convert()
with open('prediction_model.tflite', 'wb') as f:
    f.write(tflite_model)
    
p = "prediction_model.tflite"
interpreter = tf.lite.Interpreter(model_path=p)
output_details = interpreter.get_output_details()

Thanks for your help

**Comment by tiruk007** [comment]
Hi @AntoineChauviere 
Sorry for the late reply. Could you please try to save the model with signature and refer to this [document](https://www.tensorflow.org/api_docs/python/tf/saved_model/save#args).
Thank you!
